mmf/common/test_reporter.py

```diff
@@ -170,16 +161,13 @@ def add_to_report(self, report, model):
         self.report = self.report + results
 
     def reshape_and_gather(self, report, key):
-
         if key in report:
             num_dims = report[key].dim()
             if num_dims == 1:
                 report[key] = gather_tensor(report[key]).view(-1)
-
-            elif num_dims == 2:
-                _, enc_size = report[key].size()
+            elif num_dims >= 2:
+                enc_size = report[key].size(-1)
```

Comment:
Taking  `.size(-1)` will only take the last dimension. So I think this won't work when `num_dims > 2`

```suggestion
                enc_size = *report[key].size()[1:]
```

Reply:
Thanks. This has been resolved.